crates/symbolicator-proguard/src/symbolication.rs

```diff
@@ -184,19 +184,35 @@ impl ProguardService {
         frame: &JvmFrame,
         release_package: Option<&str>,
     ) -> Vec<JvmFrame> {
+        let deobfuscated_signature: Option<proguard::DeobfuscatedSignature> =
+            frame.signature.as_ref().and_then(|signature| {
+                for mapper in mappers {
+                    if let Some(deobfuscated_signature) = mapper.deobfuscate_signature(signature) {
+                        return Some(deobfuscated_signature);
+                    }
+                }
+                None
+            });
+        let params = deobfuscated_signature
+            .as_ref()
+            .map_or(String::new(), |sig| {
+                String::from_iter(sig.parameters_types().map(|param| format!("{},", param)))
+            });
```

Comment:
Might be worth figuring out how to avoid allocating intermediate `String`s here.
I would also rather propagate the `None` here, instead of having an empty `String`.

Down below you can then rather match on `params`, as that is the thing you are using.

You are using the `deobfuscated_signature` here. I thought that the parameters in the proguard file are obfuscated as well? Or are those already deobfuscated?

This could really use a small integration test with a minimized proguard file.

Reply:
The parameters stored in the returned struct `deobfuscated_signature` are already converted to proper java types and deobfuscated (when necessary). Here we really just need to concatenate them.